test/parallel/test-http-hostname-typechecking.js

```diff
@@ -18,10 +18,16 @@ vals.forEach((v) => {
   assert.throws(() => http.request({host: v}), errCheck('host'));
 });
 
+const reqs = [];
 // These values are OK and should not throw synchronously
 ['', undefined, null].forEach((v) => {
   assert.doesNotThrow(() => {
-    http.request({hostname: v}).on('error', common.noop);
-    http.request({host: v}).on('error', common.noop);
+    reqs.push(http.request({hostname: v}).on('error', common.noop));
```

Comment:
@addaleax not sure I understand.

```js
http.request({hostname: v}).on('error', common.noop).end();
```

would still work as intended no? The `error` event would still emitted.